### DISCUSSION TITLE
Official support for database ER diagrams

### DISCUSSION THREAD

**Discussion by jlynchMicron** [sqlalchemy/sqlalchemy#8840] [Usage Questions]
### Describe the use case

When designing databases, it can be very useful to visually see the tables and their relationships to verify intended changes were implemented correctly.  Could there be official SA support for visual database modeling/diagrams?

ER Modeling wiki: https://en.wikipedia.org/wiki/Entity%E2%80%93relationship_model

Potential implementation: https://github.com/Alexis-benoist/eralchemy
(Note: Broken for SA 1.4)

Working for SA 1.4: https://github.com/maurerle/eralchemy2
(Note: Seems to not work with STRUCT data type for sqlalchemy-bigquery dialect)

### Databases / Backends / Drivers targeted

All

### Example Use

See linked examples

**Comment by jlynchMicron** [comment]
Looks like this tool also exists.  Not sure if its still officially supported.
https://github.com/sqlalchemy/sqlalchemy/wiki/SchemaDisplay

**Comment by zzzeek** [comment]
I would look around on pypi, a google for "sqlalchemy diagram pypi" found quite a bit.
![Screenshot from 2022-11-18 15-56-07](https://user-images.githubusercontent.com/128223/202800899-6d4a72ea-1088-4039-9a4b-5cdd1d7940bc.png)

**Comment by jlynchMicron** [comment]
> > I would look around on pypi, a google for "sqlalchemy diagram pypi" found quite a bit. ![Screenshot from 2022-11-18 15-56-07](https://user-images.githubusercontent.com/128223/202800899-6d4a72ea-1088-4039-9a4b-5cdd1d7940bc.png)

I have been looking around, but they all seem to have spotty support for one reason or another, especially with non-common dialects like BigQuery.  I was wondering if there was core SA support, maybe visualization support would work for all "properly implemented" dialects.

**Reply by jlynchMicron** [reply]
I have tried both ERAlchemy and SchemaDisplay, and they both break for sqlalchemy-bigquery dialect

**Reply by zzzeek** [reply]
there should not be any linkage between a specific dialect and an ER display tool, unless you are relying upon the dialect's reflection capabilities which might not be present for bigquery.   That issue would be present no matter what.
